### ISSUE TITLE
Add support for additional NFS file system in DTS

### ISSUE THREAD

**Issue by miczyg1** [Dasharo/dasharo-issues#366]
**The problem you're addressing (if any)**
Reliable HCL reports depend on the availability of the drivers in Linux kernel and initrd. It is often impossible to get complete HCL reports because of a missing driver. It is also not desirable to bloat the kernel image and initrd anymore.

**Describe the solution you'd like**

A public, read-only NFS specially made for DTS to store user-space software, libraries and additional kernel modules.
The NFS mountpoint could be in the initrd fstab as optional.

**Where is the value to a user, and who might that user be?**
Sometimes it is not possible to send a report because of missing network drivers, or HCL does not contain information about certain peripherals. DTS could also become a good basic test OS for various system components. For example for various GPUs on desktops if the drivers were there. It would eliminate the necessity of installing Ubuntu or other Linux distro, just to test something quickly.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by miczyg1** [comment]
@pietrushnic cc

**Comment by mkopec** [comment]
I'm not sure if NFS is the best choice to use over a WAN link. It's unencrypted and meant to be used over low-latency links like a LAN.

**Comment by miczyg1** [comment]
@mkopec that's just an idea for now. Your input is also valuable here to polish it.

**Comment by macpijan** [comment]
> It is often impossible to get complete HCL reports because of a missing driver.

Can we gather concrete examples of what was missing in this (or other dedicated) issue?

**Comment by miczyg1** [comment]
> > It is often impossible to get complete HCL reports because of a missing driver.
> 
> Can we gather concrete examples of what was missing in this (or other dedicated) issue?

That is not the subject of the issue. I am simply trying to show motives. But in general there might be nvidia drivers that are externally installed via dkms for example (not a part of kernel)

**Comment by macpijan** [comment]
But I am challenging the motive. So let's create a separate issue with such subject to collect real-life examples on what was missing, and see if the proposed solution is really a fix for that.

**Comment by miczyg1** [comment]
> But I am challenging the motive. So let's create a separate issue with such subject to collect real-life examples on what was missing, and see if the proposed solution is really a fix for that. Or maybe we can simply link to this issue, whenever we face such situation.

The motive also is to not bloat kernel/initrd any more. We have another request for tpm2-tools for example: https://github.com/Dasharo/dasharo-issues/issues/216 which probably have many dependencies on libs.

**Comment by pietrushnic** [comment]
@miczyg1 In meta-rte, there is some success in introducing `cpu`; do you think it would solve your issues? We all agree NFS is not a good path forward.

**Comment by pietrushnic** [comment]
@TomaszAIR @artur-rs I assigned to https://github.com/Dasharo/dasharo-issues/milestone/21, but I expect some decisions regarding this issue because I doubt it can be resolved in v1.2.13 It is rather a long-term design discussion, but we should try divide the problem and conquer piece by piece.

**Comment by miczyg1** [comment]
> @miczyg1 In meta-rte, there is some success in introducing `cpu`; do you think it would solve your issues? We all agree NFS is not a good path forward.

I'm fine with almost anything that will work. THe question is whether `cpu` will solve any missing drivers problems. What if we miss a network driver?

**Comment by pietrushnic** [comment]
@miczyg1 

> It was frequently mentioned that DTS could be run on any HW to submit a HCL as an input for our evaluation (for whatever needs, either a port, or HW ROT status or anything else).

True, but it doesn't mean we shouldn't change that assumption if supporting a wide range of hardware is not economically feasible for us. Maybe the path should be as follows:
- DTS support Dasharo Roadmap hardware
- if you want DTS to support something more, please create a feature request expressing the lack of compatibility you care about (missing ethernet driver, missing chipset driver, etc.)

> Compiling in (possibly) all network drivers will not make it blown like a distro. But network drivers are crucial. Without them DTS is not so usable for others, unless it is burnt on USB stick. Bu even then one has to send us the HCL manually, which is not user friendly.

I agree network drivers are crucial; the key question is: what drivers do we miss now, and how can we test those with the hardware we already have? If we can narrow down missing drivers that can be verified, I see no issue in adding those to DTS. That is way more [SMART](https://en.wikipedia.org/wiki/SMART_criteria) task than holistic expandable DTS design, which we can't define now.

**Comment by artur-rs** [comment]
Linked with #247, adding to the status meeting agenda.
